internal/migrate/state.go

```diff
@@ -0,0 +1,125 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package migrate
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/flow-emulator/storage/migration"
+	emulatorMigrate "github.com/onflow/flow-emulator/storage/migration"
+	"github.com/onflow/flow-emulator/storage/sqlite"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go/cmd/util/ledger/migrations"
+	"github.com/onflow/flowkit/v2"
+	"github.com/onflow/flowkit/v2/config"
+	"github.com/onflow/flowkit/v2/output"
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-cli/internal/command"
+)
+
+var stateFlags struct {
+	Contracts []string `default:"" flag:"contracts" info:"contract names to migrate"`
+	DBPath    string   `default:"./flowdb" flag:"db-path" info:"path to the sqlite database file"`
+}
+
+var stateCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "state",
+		Short:   "Migrate the state of a SQLite Flow Emulator database",
+		Example: `flow migrate state`,
+		Args:    cobra.MinimumNArgs(0),
+	},
+	Flags: &stateFlags,
+	RunS:  migrateState,
+}
+
+func migrateState(
+	args []string,
+	globalFlags command.GlobalFlags,
+	_ output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (command.Result, error) {
+	if globalFlags.Network != "emulator" {
+		return nil, fmt.Errorf("state migration is only supported for the emulator network")
+	}
+
+	contracts, err := resolveStagedContracts(state, stateFlags.Contracts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve staged contracts: %w", err)
+	}
+
+	store, err := sqlite.New(stateFlags.DBPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open database: %w", err)
+	}
+
+	rwf := &migration.NOOPReportWriterFactory{}
```

Comment:
We'll probably want some kind of writer here to provide feedback to the developer

Reply:
Sounds good, I'll see how I can surface it.

Reply:
I added a `--save-report` flag - open to feedback here about naming/acceptable defaults. Assumed it's probably best not to pollute user's workspace if not manually specified (so default to NOOP) but can change if wanted.

https://github.com/onflow/flow-cli/pull/1426/files#diff-fbaffef2f65ec644d807fb85832bf2077233decf77b14155006c99f7159cd850R43